src/renderer/qssWidget/js/qssMouseWidget.js

```diff
@@ -0,0 +1,230 @@
+                    }
+                }
+            }
+        }
+    });
+
+    /**
+     * Represents the QSS mouse toggle widget.
+     */
+    fluid.defaults("gpii.qssWidget.mouseWidgetToggle", {
+        gradeNames: ["fluid.viewComponent", "gpii.psp.selectorsTextRenderer"],
+
+        selectors: {
+            toggleButton: ".flc-toggleButton",
+            settingTitle: ".flc-qssMouseWidgetToggle-settingTitle"
+        },
+
+        enableRichText: true,
+
+        model: {
+            setting: {},
+            value: "{that}.model.setting.value",
+            messages: {
+                settingTitle: "{that}.model.setting.schema.title"
+            }
+        },
+
+        components: {
+            toggleButton: {
+                type: "gpii.psp.widgets.switch",
+                container: "{that}.dom.toggleButton",
+                options: {
+                    model: {
+                        enabled: {
+                            expander: {
+                                funcName: "gpii.qssWidget.mouseWidgetToggle.transformValue",
+                                args: ["{mouseWidgetToggle}.model.value", "{mouseWidgetToggle}.model.setting.schema"]
+                            }
+                        }
+                    },
+                    invokers: {
+                        toggleModel: {
+                            funcName: "gpii.qssWidget.mouseWidgetToggle.toggleModel",
+                            args: ["{that}", "{mouseWidgetToggle}", "{channelNotifier}.events.onQssWidgetSettingAltered"]
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    /**
+     * Transforms a number value to boolean.
+     * @param {Number} value - The value of the setting.
+     * @param {Object} schema - Describes the schema of the setting.
+     * @return {Boolean} The modified value.
+     */
+    gpii.qssWidget.mouseWidgetToggle.transformValue = function (value, schema) {
+        if (value === schema.mapOff) {
+            return false;
+        } else if (value === schema.mapOn) {
+            return true;
+        } else {
+            return value;
+        }
+    };
+
+    /**
+     * Invoked whenever the user has activated the "switch" UI element (either
+     * by clicking on it or pressing "Space" or "Enter"). What this function
+     * does is to change the `enabled` model property to its opposite value.
+     * @param {Component} that - The `gpii.psp.widgets.switch` instance.
+     * @param {Component} toggleWidget - The `gpii.qssWidget.mouseWidgetToggle` instance.
+     * @param {fluid.event} event - onQssWidgetSettingAltered event
+     */
+    gpii.qssWidget.mouseWidgetToggle.toggleModel = function (that, toggleWidget, event) {
+        if (toggleWidget.model.setting.schema.mapOff || toggleWidget.model.setting.schema.mapOn) {
+            if (that.model.enabled) {
+                toggleWidget.model.setting.value = toggleWidget.model.setting.schema.mapOff;
```

Comment:
These updates need to be made via the ChangeApplier API or a model relay, rather than adjusting the model in place